src/storage/mutate/AddVerticesProcessor.cpp

```diff
@@ -92,6 +94,20 @@ void AddVerticesProcessor::doProcess(const cpp2::AddVerticesRequest& req) {
                 }
 
                 auto key = NebulaKeyUtils::vertexKey(spaceVidLen_, partId, vid, tagId);
+                if (ifNotExists_) {
+                    if (!visited.emplace(key).second) {
+                        continue;
+                    }
+                    auto obsIdx = findOldValue(partId, vid, tagId);
```

Comment:
Maybe we should emplace result into vertexCache_ in `findOldValue`, at least add a todo

Reply:
make sense

Reply:
Do you mean vetexCache_->evict(vid,tagId) if (vid,tagId) already exists in kvstroe In the findOldValue function?, i'm a little confused

Reply:
Sorry for late reply, I mean we could insert to cache if we find an old value. The logic could be handled in another PR,  the logic is weird indeed.

Reply:
It's okay, I understand what you mean